include/cjdb/contracts.hpp

```diff
@@ -4,10 +4,46 @@
 #ifndef CJDB_CONTRACTS_HPP
 #define CJDB_CONTRACTS_HPP
 
-#include <cstdio>
 #include <string_view>
 #include <type_traits>
 
+#ifdef _MSC_VER
+	#define CJDB_PRETTY_FUNCTION __FUNCSIG__
+	#define CJDB_FORCE_INLINE __forceinline
+#else
+	#define CJDB_PRETTY_FUNCTION __PRETTY_FUNCTION__
+	#define CJDB_FORCE_INLINE [[gnu::always_inline]] inline
+#endif // _MSC_VER
+
+#ifndef CJDB_PRINT_ERROR
```

Comment:
What is this guard protecting against?

Reply:
My idea is that the user could optionally define their own `CJDB_PRINT_ERROR` function.

Reply:
`cjdb::contracts_detail` is a detail namespace, so a user can't do this under the proposed model. If you want to make this configurable, please put it in namespace `cjdb`. I think the macro should probably be opt-out, not opt-in? Otherwise folks won't get any output in their debug builds, which would be surprising.

Reply:
My idea is that the user could define or choose an appropriate function under whichever namespace, then define the macro `CJDB_PRINT_ERROR` to call that function. The `cjdb::contracts_detail::print_error` function is only created when `CJDB_PRINT_ERROR` is _not_ already defined.

There may of course be better ways of making this configurable.

Reply:
Could you please have a think about a more C++-oriented way to configure this? You don't need to implement it just yet: come back here and post the idea, but if we can avoid a macro, that'd be great.

Reply:
I changed the implementation to use a function pointer `cjdb::print_error` which the user can set to something else. I still do have two macro configuration options `CJDB_USE_IOSTREAM` (as you know, `#include <iostream>` introduces some overhead, so should be opt-in) and `CJDB_SKIP_STDIO`, in which case the user must provide their own definition for `cjdb::print_error`.